File: src/MaksIT.CertsUI/Models/APIKeys/Search/SearchAPIKeyResponse.cs
using MaksIT.Core.Abstractions.Webapi;

namespace MaksIT.CertsUI.Models.APIKeys.Search;

public class SearchAPIKeyResponse : ResponseModelBase {
  public Guid Id { get; set; }
  public DateTime CreatedAt { get; set; }
  public string? Description { get; set; }
  public DateTime? ExpiresAt { get; set; }

  public bool IsGlobalAdmin { get; set; }
}

File: src/MaksIT.WebUI/src/models/certsUI/apiKeys/search/SearchAPIKeyResponse.ts
import { ResponseModelBase } from '../../../ResponseModelBase'

/** Mirrors <c>MaksIT.CertsUI.Models.APIKeys.Search.SearchAPIKeyResponse</c>. */
export interface SearchAPIKeyResponse extends ResponseModelBase {
  id: string
  createdAt: string
  description?: string | null
  expiresAt?: string | null
  isGlobalAdmin: boolean
}
